### PR TITLE
Fix: Validators - porting in Swift + refactoring [utils]

### PR DESCRIPTION
**Pull request by nicorsm**
## What's new in this PR?

Ported code of validators (string length, email, password, etc.) to Swift. Still needs to fix tests.

### CODE REVIEW

File: Source/Validation/ZMAccentColorValidator.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+//
+//  ZMAccentColorValidator.swift
+//  WireUtilities
+//
+//  Created by Nicola Giancecchi on 09.08.19.
+//
```
Comment (billypchan) [inline Source/Validation/ZMAccentColorValidator.swift]: remember to update the boilplate :)

File: Source/Validation/ZMAccentColorValidator.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import UIKit
+
+class ZMAccentColorValidator: ZMPropertyValidator {
+
+    static func validateValue(_ ioValue: inout Any?) throws -> Bool {
+        
+        guard let value = ioValue as? Int16,
+            value < ZMAccentColor.min.rawValue,
+            ZMAccentColor.max.rawValue < value else {
+            return false
```
Comment (billypchan) [inline Source/Validation/ZMAccentColorValidator.swift]: The original method returns true for all cases. This method should return false when the color is invalid.

File: Source/Validation/ZMPhoneNumberValidator.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        if finalPhoneNumber as! NSString != phoneNumber {
+            ioValue = finalPhoneNumber
+        }
+
+        return false
+    }
+    
+    static func isValidPhoneNumber(_ phoneNumber: String) -> Bool {
+        var phoneNumber: Any? = phoneNumber
+        return try! validateValue(&phoneNumber)
```
Comment (typfel) [inline Source/Validation/ZMPhoneNumberValidator.swift]: This force try will crash if `validateValue` throws an error.

File: Source/Validation/ZMEmailAddressValidator.swift [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        if emailAddress != ioValue as? NSString? {
+            ioValue = emailAddress
+        }
+        
+        return false
+    }
+    
+    static func isValidEmailAddress(_ emailAddress: String) -> Bool {
+        var emailAddress: Any? = emailAddress
+        return try! validateValue(&emailAddress)
```
Comment (typfel) [inline Source/Validation/ZMEmailAddressValidator.swift]: This force try will crash if `validateValue` throws an error.

File: Source/Validation/ZMPhoneNumberValidator.swift [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+
+import UIKit
+
+public class ZMPhoneNumberValidator: ZMPropertyValidator {
+
+    public static func validateValue(_ ioValue: inout Any?) throws -> Bool {
```
Comment (billypchan) [inline Source/Validation/ZMPhoneNumberValidator.swift]: I found we have a library for validate a phone number, we may consider no need to write them here:
https://github.com/wireapp/libPhoneNumber-iOS

File: Source/Public/StringLengthValidator.swift [hunk 6]
```diff
@@ -29,20 +29,19 @@ import Foundation
...
         
         static let tooLong = StringLengthError(domain: ZMObjectValidationErrorDomain,
-                                               code: Int(ZMManagedObjectValidationErrorCode.objectValidationErrorCodeStringTooLong.rawValue),
+                                               code: ZMManagedObjectValidationErrorCode.tooLong.rawValue,
                                                userInfo: nil)
     }
     
-    @objc(validateValue:minimumStringLength:maximumStringLength:maximumByteLength:error:)
-    static public func validateValue(_ ioValue: AutoreleasingUnsafeMutablePointer<AnyObject?>,
+    static public func validateValue(_ ioValue: inout Any?,
```
Comment (billypchan) [inline Source/Public/StringLengthValidator.swift]:
```suggestion
    @discardableResult static public func validateValue(_ ioValue: inout Any?,
```

Comment (billypchan) [reply Source/Public/StringLengthValidator.swift]: And we should add `@discardableResult` for similar methods.

File: Source/Validation/ZMPhoneNumberValidator.swift (hunk 5 above)
Comment (typfel) [reply Source/Validation/ZMPhoneNumberValidator.swift]: We can look into it in a separate PR. We don't want change any behaviour in this PR.

**Comment by nicorsm** [comment]
@zenkins release major

**Pull request merged**